server/scripts/populate.py

```diff
@@ -0,0 +1,150 @@
+import psycopg2
+import yaml  # noqa
+import uuid
+import pprint
+from datetime import datetime
+from os import path
+from configparser import ConfigParser
+
+filepath = "clubs.yml"
+pp = pprint.PrettyPrinter(indent=2)
+club_non_nullable = ["name", "description",
+                     "size", "banner_image", "icon_image"]
+event_non_nullable = ["name", "description",
+                      "banner_image", "start_time", "end_time"]
+tags_enum = ["Community", "Tech", "Creative", "Active", "Volunteering", "Gaming",
+             "Career", "Engineering", "Science", "Environment", "Arts", "Math", "Health"]
+tag_entities = {}
+
+# Assumptions: each club has its own events. Many-to-many not supported
+# yet. See example clubs.yml file for example input. db.ini file must be
+# in same directory
+
+def insert_tags(): 
+    conn = connect()
+    try:
+        with conn.cursor() as cursor:
+            for tag in tags_enum:
+                _id = str(uuid.uuid4())
+                tag_entities[tag] = _id
+                cursor.execute("INSERT INTO tag VALUES (%s,%s,%s,%s);",
+                               (_id, datetime.now(), datetime.now(), tag))
+        conn.commit()
+        print("Tags Inserted")
+    finally:
+        conn.close()
+
+
+def club_generator(clubs):
+    for club in clubs:
+        yield club
+
+
+def read_clubs(filepath):
+    with open(filepath) as file:
+        clubList = yaml.load(file, Loader=yaml.FullLoader)
+        for club in clubList:
+            if not verify(club, club_non_nullable, "club"):
+                return None
+        return club_generator(clubList)
+
+
+def verify(entity, fields, typeof):
+    for field in fields:
+        if entity.get(field) is None:
+            name = "Untitled" if entity.get("name") is None else entity.get("name")
+            print("Error:", typeof, name,
+                  "is missing mandatory field", field)
+            return False
+        events = entity.get("events")
+        if events is not None:
+            for event in events:
+                if not verify(event, event_non_nullable, "event"):
+                    return False
+        tags = entity.get("tags")
+        if tags is not None:
+            for tag in tags:
+                if tag not in tags_enum:
+                    print("Error:", typeof, entity.get(
+                        "name"), "has illegal tag", tag)
+                    return False
+    return True
+
+
+def insert_clubs():
+    club_list = read_clubs(filepath)
+    if club_list is None:
+        print("Something failed; aborting...")
+        return
+    conn = connect()
```

Comment:
This is fine in our limited use case, but in general we want to avoid calling potentially-expensive functions like `connect` as much as possible.

`connect` is called in both `insert_tags` and `insert_clubs` to get access to the same database both times. Is it possible for the two functions to share the same connection?